src/SharpLearning.Backend.Cntk.Test/CntkRawMnistTest.cs

```diff
@@ -0,0 +1,585 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.IO.Compression;
+using System.Net;
+using CNTK;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+// To avoid name-clash with SharpLearning.Backend.DataType.
+using CntkDataType = CNTK.DataType;
+using System.Text;
+
+namespace SharpLearning.Backend.Cntk.Test
+{
+    [TestClass]
+    public class CntkRawMnistTest
+    {
+        const string DownloadPath = "MnistTest";
+
+        [TestMethod]
+        public void MnistTrainTest()
+        {
+            // Data set
+            var mnist = Mnist.Load(DownloadPath);
+            var trainingSampleCount = mnist.TrainImages.Length;
+            var dataShape = new int[] { 28, 28, 1 };
+            var numberOfClasses = 10;
+            var outputValidationError = false;
+
+            // Set global data and device types. 
+            var dataType = CntkDataType.Float;
+            var device = DeviceDescriptor.UseDefaultDevice();
+            Trace.WriteLine("Using device: " + device.Type);
+            // These are just hardcoded in the helper class.
+            CntkLayers.DataType = dataType;
+            CntkLayers.Device = device;
+
+            // Define small convolutional neural network.
+            var net = CntkLayers.Input(dataShape);
+
+            net = CntkLayers.Conv2D(net, 3, 3, 32);
```

Comment:
@mdabros I would much rather have the raw CNTK functions here, the whole point is to function as an example of the raw CNTK methods. Not to wrap stuff. Wrapping stuff is for the backend, this is to see quickly how the raw API looks and works, wrapping does not help, it just hides what is really the purpose here.

Reply:
@nietras I will make the network simpler, just a logistic regression, and use the operators directly. I will keep the helper class CntkLayer in case we want to experiment with more advanced networks. As you mention, the functions on the class simply wraps a set of cntk operators for creating what corresponds to layer operations. So it also serves as an example of the minimal set of operators we need to support.